hydromt/components/table.py

```diff
@@ -0,0 +1,133 @@
+"""Table component."""
+
+import glob
+import os
+from os.path import basename, dirname, join
+from typing import (
+    TYPE_CHECKING,
+    Dict,
+    Optional,
+    Union,
+)
+
+import pandas as pd
+
+from hydromt.components.base import ModelComponent
+from hydromt.hydromt_step import hydromt_step
+from hydromt.utils.constants import DEFAULT_TABLE_FILENAME
+
+if TYPE_CHECKING:
+    from hydromt.models.model import Model
+
+__all__ = ["TableComponent"]
+
+
+class TableComponent(ModelComponent):
+    """Table Component."""
+
+    def __init__(
+        self,
+        model: "Model",
+    ):
+        """Initialize a TableComponent.
+
+        Parameters
+        ----------
+        model: Model
+            HydroMT model instance
+        """
+        self._data: Optional[Dict[str, Union[pd.DataFrame, pd.Series]]] = None
+        super().__init__(model=model)
+
+    @property
+    def data(self) -> Dict[str, Union[pd.DataFrame, pd.Series]]:
+        """Model tables."""
+        if self._data is None:
+            self._initialize_tables()
+
+        assert self._data is not None
```

Comment:
it is not for functionality as this will always be true, it is just to help automated checkers like mypy make deductions it can't otherwise make. I don't really understand why including it is something that you are worried about.

Reply:
It became more clear with the mypy discussion we had. What makes me worried is more the burden on the developer to support type checking. I doubt plugins will do all this (mypy, pydantic, type_checking) unless they get more capacity/budget.